app/src/main/java/de/grobox/transportr/trips/detail/StopViewHolder.kt

```diff
@@ -42,6 +43,9 @@ internal class StopViewHolder(v: View, private val listener : LegClickListener)
         if (stop.arrivalTime != null) {
             setArrivalTimes(fromTime, fromDelay, stop)
         } else {
+            // ensure fields for stops without times have a default width, so that stop circles align
+            fromTime.minWidth = fromTime.paint.measureText(getTime(context, Date())).toInt()
```

Comment:
This feels a bit hacky. I feel like there should be a ConstraintLayout solution for this.

Reply:
True, feels odd, but I didn't find any solution with ContraintLayout: I think you can not reference a view outside the current list item and if there are no times given, you can't guess the width used in other list items. `minWidth` isn't an option neither as the width depends heavily on the time format ("10:00 PM" is much larger than "10:00" for example).

But if you find any other solution, that would be great!

Reply:
An alternative would be to set a dummy time in this case and set the TextView to invisible, perhaps a little less hacky.

Reply:
Right, I had forgotten about the problem of different line items. The invisible dummy time sounds a bit better then the measure approach.

Just make sure the `fromTime` is set to visible when we have a stop time, otherwise it will disappear when the line item view gets recycled with a stop time.